app/color.py

```diff
@@ -30,12 +30,8 @@ def get(colorType, delta=0):
   else:
     colorIndex = app.prefs.color[colorType]
   colorIndex = min(colors - 1, colorIndex + delta)
-  #colorIndex = colorIndex % colors
-  r = cache__.get(colorIndex)
-  if r is not None:
-    return r
-  color = curses.color_pair(colorIndex)
+  color = cache__.get(colorIndex) or curses.color_pair(colorIndex)
+  cache__[colorIndex] = color
   if colorType in ('error', 'misspelling'):
     color |= curses.A_BOLD | curses.A_REVERSE
```

Comment:
What is better about the new code?
At first glance it looks like it will repeatedly reset the same cache value; and repeatedly check for 'error' or 'misspelling'.

Reply:
Accidentally put the answer to this question below.

Reply:
Thanks for explaining. I'm going to go ahead with this PR, but I'll likely want to change this code again. Here's why: in some coding, avoiding if-statements is important for performance, in other situations, avoiding function calls is more important.

On one extreme, in assembly on a Cell processor, like in the PlayStation 3, avoiding if-statements (conditional-branching) is critical for high performance; while a calling a function (unconditional-branch) is close to free by comparison.

On another extreme, in Python (and many other scripting languages) avoiding function calls is critical for high performance; while if-statements are close to free by comparison.

In many environments, (in both extremes above) calling out to an external (non-local) library is very time consuming. One of the main reasons for color.get() to exist is to reduce the number of calls out to curses.color_pair() since calling into curses from Python is time consuming.